packages/flutter/lib/src/painting/cupertino_rounded_rectangle_border.dart

```diff
@@ -0,0 +1,504 @@
+        ..cubicTo(rightX(0), bottomY(0.68440652),
+            rightX(0.68440676), bottomY(0),
+            rightX(1.52866495), bottomY(0))
+        ..cubicTo(rightX(1.52866495), bottomY(0),
+            rightX(1.52866495), bottomY(0),
+            rightX(1.52866495), bottomY(0))
+        ..cubicTo(rightX(1.52866495), bottomY(0),
+            rightX(1.52866495), bottomY(0),
+            rightX(1.52866495), bottomY(0))
+        ..lineTo(centerX, bottomY(0))
+        ..cubicTo(leftX(1.52866483), bottomY(0),
+            leftX(1.52866483), bottomY(0),
+            leftX(1.52866483), bottomY(0))
+        ..lineTo(leftX(1.52866471), bottomY(0))
+        ..cubicTo(leftX(0.68440646), bottomY(0),
+            leftX(-0.00000004), bottomY(0.68440676),
+            leftMidX(0), centerY)
+        ..cubicTo(leftMidX(0), centerY,
+            leftMidX(0), centerY,
+            leftMidX(0), centerY)
+        ..cubicTo(leftMidX(0), centerY,
+            leftMidX(0), centerY,
+            leftMidX(0), centerY)
+        ..lineTo(leftMidX(0), centerY)
+        ..cubicTo(leftMidX(0), centerY,
+            leftMidX(0), centerY,
+            leftMidX(0), centerY)
+        ..lineTo(leftMidX(0), centerY)
+        ..cubicTo(leftX(0.00000007), topY(0.68440652),
+            leftX(0.68440664), topY(-0.00000001),
+            leftX(1.52866483), topY(0))
+        ..cubicTo(leftX(1.52866483), topY(0),
+            leftX(1.52866483), topY(0),
+            leftX(1.52866483), topY(0))
+        ..lineTo(centerX, topMidY(0))
+        ..close();
+    }
+
+    // The radius multiplier where the resulting shape will perfectly concave at
+    // any dimension.
+    const double maxMultiplier = 3.0573;
+
+    // The multiplier used to describe the minimum radius for the dynamic shape
+    // round option.
+    const double dynamicShapeMinMultiplier = 0.76435;
+
+    // The multiplier used to describe the minimum radius for the dynamic radius
+    // round option.
+    const double dynamicRadiusMinMultiplier = 2.0;
+
+    // The edge length at which the corner radius must be at its maximum so as to
+    // maintain a the appearance of a completely concave shape.
+    const double minRadiusEdgeLength = 200.0;
+
+    switch(mode) {
+      case RoundedRectCornerMode.dynamicShape:
+        final double min = math.min(radius, math.min(rrect.width, rrect.height));
+        limitedRadius = min * dynamicShapeMinMultiplier;
+        if (width > maxMultiplier * radius && height > maxMultiplier * radius)
+          return roundedRect1();
+        else if (width > maxMultiplier * radius)
+          return roundedRect2a();
+        else if (height > maxMultiplier * radius)
+          return roundedRect2b();
+        else if (height > width)
+          return roundedRect3a();
+        return roundedRect3b();
+
+      case RoundedRectCornerMode.dynamicRadius:
+        final double min = math.min(rrect.width, rrect.height);
+
+        // As the minimum side edge length (where the round is occurring)
```

Comment:
I didn't really understand this either. Is the side edge length approaching 0 or side edge length - radius * 2? Feel free to add more text to elaborate and define each term being used.